tests/commands/test_store_client.py

```diff
@@ -0,0 +1,321 @@
+# Copyright 2020 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+import json
+import logging
+import os
+from http.cookiejar import MozillaCookieJar, Cookie
+from unittest.mock import patch
+
+import pytest
+from macaroonbakery import httpbakery
+
+from charmcraft.cmdbase import CommandError
+from charmcraft.commands.store.client import Client, BASE_URL, _AuthHolder, visit_page_with_browser
+
+
+# --- AuthHolder tests
+
+@pytest.fixture
+def auth_holder(tmp_path):
+    """Produce an _AuthHolder instance with encapsulated resources.
+
+    In detail:
+
+    - writes into a temp dir
+
+    - with a mocked webbrowser module
+    """
+
+    ah = _AuthHolder()
+    ah._cookiejar_filepath = str(tmp_path / 'test.credentials')
```

Comment:
any reason we couldn't pass this in, instead of patching it after the fact?

Reply:
It would be something the code would instrument only to the test.

We would need to test what happens if it receives it or if it doesn't, etc.